src/cli/playbook/input.rs

```diff
@@ -0,0 +1,154 @@
+use crate::ff::Field;
+
+use std::any::type_name;
+use std::fs::File;
+use std::io;
+use std::io::{stdin, BufRead, BufReader, Read};
+
+use std::path::PathBuf;
+
+pub trait InputItem {
+    fn from_str(s: &str) -> Self;
+}
+
+impl<F: Field> InputItem for F {
+    fn from_str(s: &str) -> Self {
+        let int_v = s.parse::<u128>().unwrap();
+        F::from(int_v)
+    }
+}
+
+impl InputItem for u64 {
+    fn from_str(s: &str) -> Self {
+        s.parse::<u64>().unwrap()
+    }
+}
+
+impl<I: InputItem> InputItem for (I, I) {
+    fn from_str(s: &str) -> Self {
+        let mut iter = s.split(',');
```

Comment:
I am not sure how I feel about rejecting the input `10,20,\n` - maybe allow some sloppiness while entering the input?

Reply:
i think you would want to try to avoid sloppiness in placement of `\n`. i could imagine accidentally doing something like

`"10,20,30\n40"`

when you intended

`"10,20\n30,40"`

or something along those lines. i guess that would get caught later when the total count doesn't add up, but your error message would be better if you capture it here.

it's not a big deal tho. the error would get caught either way

Reply:
good point about better error messages. I think we should implement it your way, so I changed it to use `split_once`